asyncio_run_in_process/_child.py

```diff
@@ -178,7 +167,28 @@ async def _do_async_fn(
             raise Exception("unreachable")
 
 
-def _run_process(parent_pid: int, fd_read: int, fd_write: int) -> None:
+def _run_on_asyncio(async_fn: TAsyncFn, args: Sequence[Any], to_parent: BinaryIO) -> None:
+    loop = asyncio.get_event_loop()
+    try:
+        result: Any = loop.run_until_complete(_do_async_fn(async_fn, args, to_parent, loop))
+    except BaseException:
+        exc_type, exc_value, exc_tb = sys.exc_info()
+        # `mypy` thinks that `exc_value` and `exc_tb` are `Optional[..]` types
+        if exc_type is asyncio.CancelledError:
+            exc_value = ChildCancelled(*exc_value.args)  # type: ignore
+        remote_exc = RemoteException(exc_value, exc_tb)  # type: ignore
+        finished_payload = pickle_value(remote_exc)
+        raise
+    else:
+        finished_payload = pickle_value(result)
+    finally:
+        # XXX: The STOPPING state seems useless as nothing happens between that and the FINISHED
```

Comment:
This could be classified as YAGNI, I think when I modeled the lifecycle I expected to need something to happen there but ended up not needing it.

Reply:
I'll submit a follow-up PR to change that, as I'd rather not include any unrelated changes here